src/main/2.0/scala/com/holdenkarau/spark/testing/DataFrameSuiteBase.scala

```diff
@@ -170,6 +170,16 @@ trait DataFrameSuiteBaseLike extends SparkContextProvider
     }
   }
 
+  /**
+    * Compares if two [[DataFrame]]s are equal without caring about order of rows, by
+    * finding elements in one DataFrame not in the other. The resulting DataFrame
+    * should be empty inferring the two DataFrames have the same elements.
+    */
+  def assertDataFrameNoOrderEquals(expected: DataFrame, result: DataFrame) {
```

Comment:
Love this functionality, however I think it won't handle duplicate elements super well. If we look the documentation for except on Datasets http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.sql.Dataset@except(other:org.apache.spark.sql.Dataset[T]):org.apache.spark.sql.Dataset[T] is a little vague but we can test this quickly with:

`scala> case class A(a: Int)
defined class A

scala> val boop1 = List(A(1), A(1), A(2))
boop1: List[A] = List(A(1), A(1), A(2))

scala> val boop2 = List(A(1), A(2))
boop2: List[A] = List(A(1), A(2))

scala> session
<console>:24: error: not found: value session
       session
       ^

scala> spark\
<console>:24: error: value \ is not a member of org.apache.spark.sql.SparkSession
       spark\
            ^

scala> spark
res2: org.apache.spark.sql.SparkSession = org.apache.spark.sql.SparkSession@65698020

scala> val df1 = spark.createDataFrame(boop1)
2018-04-13 11:59:46 WARN  ObjectStore:568 - Failed to get database global_temp, returning NoSuchObjectException
df1: org.apache.spark.sql.DataFrame = [a: int]

scala> 

scala> val df2 = spark.createDataFrame(boop2)
df2: org.apache.spark.sql.DataFrame = [a: int]

scala> df1.except(df2).collect()
res3: Array[org.apache.spark.sql.Row] = Array()                                 

scala> df2.except(df1).collect()
res4: Array[org.apache.spark.sql.Row] = Array()  `